main.cu

```diff
@@ -174,144 +133,187 @@ __device__ inline int getTreeHeight(int x, int z) {
 #define SIZE_Z CONST_CEIL(UPPER_Z - LOWER_Z + 1)
 #define TOTAL_WORK_SIZE (SIZE_X * SIZE_Z)
 
+#define WORK_UNIT_SIZE (1LL << 23)
+#define BLOCK_SIZE 256
+
+/* TREE/WATERFALL CONSTANTS */
+#define WATERFALL_X 9
+#define WATERFALL_Y 76
+#define WATERFALL_Z 10
+
+#define TREE_X (WATERFALL_X - 5)
+#define TREE_Z (WATERFALL_Z - 8)
+#define TREE_HEIGHT 5
+
+#define OTHER_TREE_COUNT 2
```

Comment:
I'll have to do a more through analysis tomorrow, profiler is playing up.
I mainly added the flag system for consistency, seeing as I thought it would be merged with master, if I get the profiler working again and it is faster, I'll submit another PR to add the flags in.